src/controllers/userController.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable camelcase */
+import userService from '../services/userservice';
+import Util from '../utils/response';
+import { jwtSignUser } from '../utils/index';
+import { hashPassword } from '../helpers/index';
+
+
+const util = new Util();
+/**
+ * @param { class } provide response to user signup activity.
+ */
+class userController {
+  /**
+ * @param {req} req that contains the req body object.
+ * @param {res} res content to be rendered.
+ * @returns {object} Success or failure response on getting all users
+ */
+  static async getUsers(req, res) {
+    try {
+      const allUsers = await userService.getAllUser();
+      if (allUsers.length > 0) {
+        util.setSuccess(200, 'Users retrieved', allUsers);
+      } else {
+        util.setSuccess(200, 'No Users found');
+      }
+      return util.send(res);
+    } catch (error) {
+      util.setError(400, error);
+      return util.send(res);
+    }
+  }
+
+  /**
+ * @param {req} req that contains the req body object.
+ * @param {res} res content to be rendered.
+ * @returns {object} Success or failure response on adding a specific user
+ */
+  static async addUser(req, res) {
+    const { user, emailToken } = req;
+    try {
+      const hashpassword = await hashPassword(user.password);
+      user.password = hashpassword;
+      const {
+        id, email, firstName, lastName,
+      } = await userService.addUser(user);
+      const token = await jwtSignUser(id);
+      util.setSuccess(201, 'user Added!', {
+        token, emailToken, id, email, firstName, lastName,
+      });
+      return util.send(res);
+    } catch (error) {
+      if (error.original.routine === '_bt_check_unique') {
+        util.setError(400, 'Email already exist');
```

Comment:
Why use 400 status code? This should be 409.

Reply:
Alright